app/controllers/orders_controller.rb

```diff
@@ -74,23 +114,27 @@ def create
       @user_id = session[:user_id] if session[:user_id]
       @billing_id = @billing.id
       @order_number = order_number
-      @order = Order.new(status: "pending", total: total_order_revenue(@cart_items), confirmation_date: Time.now, order_number: @order_number, billing_id: @billing_id, user_id: @user_id)
+      @order = Order.new(status: "pending", total: total_order_revenue(@cart_items), confirmation_date: Time.now, order_number: @order_number, billing_id: @billing_id, user_id: @user_id, carrier_type: @carrier_type, carrier_price: @carrier_price)
       if @order.save
         @cart_items.each do |item|
           @order_item = OrderItem.new(quantity: item.quantity, name: item.product.name, price: item.product.price*item.quantity, status: "pending", order_id: @order.id, product_id: item.product.id)
           @order_item.save
           item.destroy
           @order_item.product.update(inventory: @order_item.product.inventory - @order_item.quantity)
         end
+
+
         if current_user
           redirect_to user_order_path(current_user.id, @order.id)
         else
           redirect_to checkout_confirmation_path(@order.id)
         end
       else
+
         render "/billings/new"
       end
     else
+
         render "/billings/new"
     end
```

Comment:
Since the end result if the use / billing haven't been saved is the same - you could render the "/billings/new" only once just before the last end